include/libtorrent/torrent_info.hpp

```diff
@@ -580,7 +573,7 @@ namespace libtorrent
 		// this is a copy of the info section from the torrent.
 		// it use maintained in this flat format in order to
 		// make it available through the metadata extension
-		boost::shared_array<char> m_info_section;
+		std::unique_ptr<char[]> m_info_section;
```

Comment:
this is where the metadata is owned. I'm saying that I believe this has to support shared ownership, so that the client asking for it can know that it will be valid still, immediately after it's returned. If the returned buffer is tied to the lifetime of this torrent object, especially if the torrent object is "evicted" or unloaded, it will become a dangling pointer.